cukinia

```diff
@@ -680,6 +681,73 @@ _cukinia_systemd_failed()
 	fi
 }
 
+# _cukinia_gpio_libgpiod: Check gpio configuration via libgpiod
+# arg1: (optional) the pins configured in input
+# arg2: (optional) the pins configured in output high
+# arg3: (optional) the pins configured in output low
+# arg4: (optional) the gpiochip used (default:gpiochip0)
+_cukinia_gpio_libgpiod()
+{
+	local input_pins
+	local output_low_pins
+	local output_high_pins
+	local gpiochip="gpiochip0"
+	local direction
+	local value
+
+	while [ $# -gt 0 ]; do
+		case "$1" in
+		-i)
+			input_pins="$2"
+			shift
+			;;
+		-l)
+			output_low_pins="$2"
+			shift
+			;;
+		-h)
+			output_high_pins="$2"
+			shift
+			;;
+		-g)
+			gpiochip="$2"
+			shift
+			break
+			;;
+		esac
+		shift
+	done
+
+	_cukinia_prepare "Checking if gpio pins are well configured via libgpiod"
+	if [ -n "$input_pins" ]; then
+		for pin in $input_pins; do
+			direction=$(gpioinfo "$gpiochip" | egrep "line +$pin:" | sed 's/ \+/\//g' | cut -d '/' -f5)
+			if [[ "input" != "$direction" ]]; then
+				return 1
+			fi
+		done
+	fi
+	if [ -n "$output_low_pins" ]; then
+		for pin in $output_low_pins; do
+			direction=$(gpioinfo "$gpiochip" | egrep "line +$pin:" | sed 's/ \+/\//g' | cut -d '/' -f5)
+			value=$(gpioinfo "$gpiochip" | egrep "line +$pin:" | sed 's/ \+/\//g' | cut -d '/' -f6)
+			if [[ "output/active-low" != "$direction/$value" ]]; then
+				return 1
+			fi
+		done
+	fi
+	if [ -n "$output_high_pins" ]; then
+		for pin in $output_high_pins; do
+			direction=$(gpioinfo "$gpiochip" | egrep "line +$pin:" | sed 's/ \+/\//g' | cut -d '/' -f5)
+			value=$(gpioinfo "$gpiochip" | egrep "line +$pin:" | sed 's/ \+/\//g' | cut -d '/' -f6)
+			if [[ "output/active-high" != "$direction/$value" ]]; then
```

Comment:
restrict the syntax to posix constructs (see "man dash" for example) for the `[[`